responder/config_sync.go

```diff
@@ -0,0 +1,102 @@
+package responder
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/flanksource/commons/logger"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+)
+
+func StartConfigSync() {
+	for {
+		logger.Infof("Syncing configuration")
+		syncConfig()
+
+		time.Sleep(1 * time.Hour)
+	}
+}
+
+func upsertConfig(configType, externalID, name, config string) error {
+
+	dbInsertConfigQuery := `INSERT INTO config_items (config_type, name, external_id, config) VALUES (?, ?, ARRAY[?], ?)`
+	dbUpdateConfigQuery := `UPDATE config_items SET config = ? WHERE external_id = ARRAY[?] AND config_type = ?`
+
+	query := db.Gorm.Exec(dbUpdateConfigQuery, config, externalID, configType)
+	if query.Error != nil {
+		logger.Errorf("Error updating config in database: %v", query.Error)
+		return query.Error
+	}
+
+	if query.RowsAffected == 0 {
+		if err := db.Gorm.Exec(dbInsertConfigQuery, configType, name, externalID, config).Error; err != nil {
+			logger.Errorf("Error inserting config into database: %v", err)
+			return query.Error
+		}
+	}
+
+	return nil
+}
+
+func syncConfig() {
+	var teams []api.Team
+	if err := db.Gorm.Find(&teams).Error; err != nil {
+		logger.Errorf("Error querying teams from database: %v", err)
+		return
+	}
+
+	for _, team := range teams {
+		teamSpecJson, err := team.Spec.MarshalJSON()
```

Comment:
I would move this into a function e.g. api.TeamFromV1(team.spec) or team.GetSpec()